time_tracker_api/time_entries/time_entries_repository.py

```diff
@@ -141,7 +141,37 @@ def count(
 
         return result.next()
 
-    def find_all(
+    def add_complementary_info(self, time_entries=None, max_count=None):
+        if time_entries:
+            custom_conditions = create_in_condition(time_entries, "project_id")
+            custom_conditions_activity = create_in_condition(
+                time_entries, "activity_id"
+            )
+
+            project_dao = projects_model.create_dao()
+            projects = project_dao.get_all(
+                custom_sql_conditions=[custom_conditions],
+                visible_only=False,
+                max_count=max_count,
+            )
+
+            add_project_info_to_time_entries(time_entries, projects)
+
+            activity_dao = activities_model.create_dao()
+            activities = activity_dao.get_all(
+                custom_sql_conditions=[custom_conditions_activity],
+                visible_only=False,
+                max_count=max_count,
+            )
+            add_activity_name_to_time_entries(time_entries, activities)
+
+            users = AzureConnection().users()
+            add_user_email_to_time_entries(time_entries, users)
+        elif not time_entries and len(conditions) > 1:
```

Comment:
just a question why don't use a `if` instead of `elif`

Reply:
That conditional, besides it is mutually exclusive, has another condition, so I recommend keeping the same verification.